sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/testinterface/ComRunner.scala

```diff
@@ -24,8 +24,8 @@ class ComRunner(bin: File,
                 args: Seq[String],
                 logger: Logger) {
 
-  /** Port over which we communicate with the distant program */
-  val port: Int = scala.util.Random.nextInt(1000) + 9000
+  private[this] val serverSocket = new ServerSocket(0)
+  private[this] val port         = serverSocket.getLocalPort
```

Comment:
It's better to make this socket a var, and open it right next to close to make it clear when resource is acquired/released.